raiden/network/transport/matrix/transport.py

```diff
@@ -1370,7 +1205,8 @@ def _set_room_id_for_address(self, address: Address, room_id: Optional[_RoomID]
 
         with self._account_data_lock:
             # no need to deepcopy, we don't modify lists in-place
-            _address_to_room_ids: Dict[AddressHex, List[_RoomID]] = self._client.account_data.get(
+            # type: Dict[AddressHex, List[_RoomID]]
```

Comment:
Still don't really like this. We should look if there is a way to 'explain' things to mypy. But we can do that later.

Reply:
@ulope Implemented in #3427 , as well as splitting and tests for the server selection mechanism.